app/main.py

```diff
@@ -1,3 +1,21 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Callable:
+        key = args
+
+        if key in cached_results:
+            print("Getting from cache")
+            return cached_results[key]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cached_results[key] = result
```

Comment:
There is no need for a `result` variable, write function output to the cache directly.